From 7e00964de0ae27f868726413e46f27145807a50d Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Fri, 15 Feb 2019 11:17:27 +0100
Subject: [PATCH] Do not prefix executables with $bindir in pkg-config files

This would otherwise break when using the executables to build other target
components (we need to rely on PATH containing the paths to utilities,
rather than use target paths).

Upstream-Status: Inappropriate [upstream wants the paths in .pc files]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 gio/meson.build  | 17 +++++++++--------
 glib/meson.build |  7 ++++---
 2 files changed, 13 insertions(+), 11 deletions(-)

diff --git a/gio/meson.build b/gio/meson.build
index 76d67cd..0584ff5 100644
--- a/gio/meson.build
+++ b/gio/meson.build
@@ -885,17 +885,18 @@ libgio_dep = declare_dependency(link_with : libgio,
 pkg.generate(libgio,
   requires : ['glib-2.0', 'gobject-2.0'],
   variables : [
+    'bindir=' + '${prefix}' / get_option('bindir'),
     'schemasdir=' + '${datadir}' / schemas_subdir,
     'dtdsdir=' + '${datadir}' / dtds_subdir,
     'giomoduledir=' + pkgconfig_giomodulesdir,
-    'gio=' + '${bindir}' / 'gio',
-    'gio_querymodules=' + pkgconfig_multiarch_bindir / 'gio-querymodules',
-    'glib_compile_schemas=' + pkgconfig_multiarch_bindir / 'glib-compile-schemas',
-    'glib_compile_resources=' + '${bindir}' / 'glib-compile-resources',
-    'gdbus=' + '${bindir}' /'gdbus',
-    'gdbus_codegen=' + '${bindir}' / 'gdbus-codegen',
-    'gresource=' + '${bindir}' / 'gresource',
-    'gsettings=' + '${bindir}' / 'gsettings',
+    'gio=gio',
+    'gio_querymodules=gio-querymodules',
+    'glib_compile_schemas=glib-compile-schemas',
+    'glib_compile_resources=glib-compile-resources',
+    'gdbus=gdbus',
+    'gdbus_codegen=gdbus-codegen',
+    'gresource=gresource',
+    'gsettings=gsettings',
   ],
   version : glib_version,
   install_dir : glib_pkgconfigreldir,
diff --git a/glib/meson.build b/glib/meson.build
index 9f1515b..a40e4f5 100644
--- a/glib/meson.build
+++ b/glib/meson.build
@@ -447,9 +447,10 @@ pkg.generate(libglib,
   subdirs : ['glib-2.0'],
   extra_cflags : ['-I${libdir}/glib-2.0/include'] + win32_cflags,
   variables : [
-    'glib_genmarshal=' + '${bindir}' / 'glib-genmarshal',
-    'gobject_query=' + '${bindir}' / 'gobject-query',
-    'glib_mkenums=' + '${bindir}' / 'glib-mkenums',
+    'bindir=' + '${prefix}' / get_option('bindir'),
+    'glib_genmarshal=glib-genmarshal',
+    'gobject_query=gobject-query',
+    'glib_mkenums=glib-mkenums',
     'glib_valgrind_suppressions=' + '${datadir}' /
       valgrind_suppression_file_install_subdir /
       fs.name(valgrind_suppression_file),
